src/main/java/technology/semi/weaviate/client/v1/auth/provider/NimbusAuth.java

```diff
@@ -0,0 +1,62 @@
+package technology.semi.weaviate.client.v1.auth.provider;
+
+import com.nimbusds.oauth2.sdk.AuthorizationGrant;
+import com.nimbusds.oauth2.sdk.ErrorObject;
+import com.nimbusds.oauth2.sdk.Scope;
+import com.nimbusds.oauth2.sdk.TokenErrorResponse;
+import com.nimbusds.oauth2.sdk.TokenRequest;
+import com.nimbusds.oauth2.sdk.TokenResponse;
+import com.nimbusds.oauth2.sdk.auth.ClientSecretPost;
+import com.nimbusds.oauth2.sdk.auth.Secret;
+import com.nimbusds.oauth2.sdk.http.HTTPResponse;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.token.AccessToken;
+import com.nimbusds.openid.connect.sdk.OIDCTokenResponse;
+import com.nimbusds.openid.connect.sdk.OIDCTokenResponseParser;
+import com.nimbusds.openid.connect.sdk.op.OIDCProviderMetadata;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import technology.semi.weaviate.client.Config;
+import technology.semi.weaviate.client.WeaviateClient;
+
+public class NimbusAuth extends BaseAuth {
+
+  public NimbusAuth() {
+    super();
+  }
+
+  protected WeaviateClient getAuthClient(Config config, String clientSecret, AuthorizationGrant authorizationGrant) throws Exception {
+    AuthResponse authResponse = getIdAndTokenEndpoint(config);
+    OIDCProviderMetadata providerMetadata = OIDCProviderMetadata.parse(authResponse.getConfiguration());
+    ClientID clientID = new ClientID(authResponse.getClientId());
+    Scope scopes = new Scope();
+    if (authResponse.getScopes() != null) {
+      Arrays.stream(authResponse.getScopes()).forEach(scopes::add);
+    }
+    Map<String, List<String>> customParams = new HashMap<>();
+    customParams.put("response_type", Arrays.asList("code id_token"));
+    customParams.put("response_mode", Arrays.asList("fragment"));
+    customParams.put("redirect_url", Arrays.asList(String.format("%s%s", config.getBaseURL(), OIDC_URL)));
+
+    TokenRequest tokenReq = new TokenRequest(providerMetadata.getTokenEndpointURI(),
+      new ClientSecretPost(clientID, new Secret(clientSecret)),
+      authorizationGrant,
+      scopes, null, customParams);
+
+    HTTPResponse tokenHTTPResp = tokenReq.toHTTPRequest().send();
+    TokenResponse tokenResponse = OIDCTokenResponseParser.parse(tokenHTTPResp);
+
+    if (tokenResponse instanceof TokenErrorResponse) {
+      ErrorObject error = ((TokenErrorResponse) tokenResponse).getErrorObject();
+      throw new RuntimeException(error.getDescription());
+    }
+
+    OIDCTokenResponse oidcTokenResponse = (OIDCTokenResponse) tokenResponse;
+    AccessToken accessToken = oidcTokenResponse.getOIDCTokens().getAccessToken();
+
```

Comment:
This would also need a warning if no refresh token is returned. You can reuse auth_no_refresh_token from weaviate/warnings.py